### PR TITLE
Changing error message from `contains interior mutability` to `may contain interior mutability`

### LINKED ISSUE #40313: Change error `contains interior mutability` to `may contain interior mutability`?
The following error is a bit misleading:

`the type SomeTrait + 'static contains interior mutability and a reference may not be safely transferrable 
across a catch_unwind boundary`

Since `SomeTrait` is a trait object, it seems to me `may contain (...)` is more precise.

### PR DESCRIPTION
**Pull request by gaurikholkar-zz**
Fixes #40313 . I have changed the message from `contains interior mutability` to `may contain interior mutability` for the following example
```
use std::cell::Cell;
use std::panic::catch_unwind;
fn main() {
    let mut x = Cell::new(22);
    catch_unwind(|| { x.set(23); });
}
```
which has been added as a ui test.

Also, the message [here](https://github.com/gaurikholkar/rust/blob/master/src/librustc_mir/transform/qualify_consts.rs#L666) and it's respective `compile-fail` test have been modified.

cc @nikomatsakis  @Mark-Simulacrum  @eddyb

### CODE REVIEW

**Comment by eddyb** [comment]
@gaurikholkar If you reword your PR description slightly so it contains "fixes #40313" then when this PR is merged, GitHub will close the issue automatically.

File: src/test/ui/interior-mutability/interior-mutability.rs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::cell::Cell;
+use std::panic::catch_unwind;
+fn main() {
+    let mut x = Cell::new(22);
+    catch_unwind(|| { x.set(23); });
```
Comment (eddyb) [inline src/test/ui/interior-mutability/interior-mutability.rs]: cc @alexcrichton I think this test is fine, but the name should probably contain`ref-unwind-safe`.

**Comment by eddyb** [comment]
@gaurikholkar I meant the description which currently contains "This is a fix to #40313 .", not the title, sorry.

**Review by eddyb — APPROVED** [review]
LGTM. I'm waiting for @alexcrichton to hear about the `RefUnwindSafe` trait.

**Comment by alexcrichton** [comment]
Sorry what about the `RefUnwindSafe` trait? I'm not sure what my input is requested on here...

**Comment by eddyb** [comment]
This PR rewords the error message when that trait is not implemented and adds a test for it. Do you have any objections to that? Also, what should the test be named?

**Comment by alexcrichton** [comment]
Nah seems fine to me! I have no opinions on test naming.

**Comment by eddyb** [comment]
Thanks! *shrug* guess this will do

@bors r+

**Pull request merged**